### DISCUSSION TITLE
Missing env variables running `npm run start` locally

### DISCUSSION THREAD

**Discussion by kwiat1990** [remix-run/remix#3512] [Q&A] [answered]
I wanted to check the production build of my app locally, so I have built it with `npm run build` and then ran `npm run start`. Normally, with other frameworks/libraries, it did always work. With Remix I get following error:
```bash
npm run start

> start
> remix-serve build

/Users/mateusz.kwiatkowski/Playground/remix-mantine/app/server/session.server.ts:6
  throw new Error("SESSION_SECRET must be set"); // my custom error
```

I can also see that `process.env` object in the production build doesn't contain any of variables I've defined. It has only the default one like `NODE_ENV` or `PATH`. So my question is how I can test the build locally and load my `.env` file?

**Comment by kiliman** [comment, marked as answer]
Remix only loads the .env file for you in development. It expects you to define them for production at your hosting platform. 

Anyway to load them locally on your production build, you’ll need to install `dotenv-cli` then update your start script to
```bash
dotenv remix-serve build
``` 
https://www.npmjs.com/package/dotenv-cli

**Reply by kwiat1990** [reply]
Thanks, `dotenv`package was the missing piece.

**Reply by nikolailehbrink** [reply]
Hi @kiliman, I am using React Router 7, followed your advice here but even with `dotenv-cli` installed it won't work. Created [this issue](https://github.com/remix-run/react-router/issues/13830) and maybe you can help? Would be highly appreciated!
